go/vt/sqlparser/ast.go

```diff
@@ -6497,113 +6497,116 @@ const (
 	SetScope_User        SetScope = "user"
 )
 
-// VarScopeForColName returns the SetScope of the given ColName, along with a new ColName without the scope information.
-func VarScopeForColName(colName *ColName) (*ColName, SetScope, error) {
+// VarScopeForColName returns the SetScope of the given ColName, along with a new ColName without the scope information,
+// and a boolean that indicates if the scope was explicitly specified, or inferred.
+func VarScopeForColName(colName *ColName) (*ColName, SetScope, bool, error) {
```

Comment:
No tests of this function? 😬

Reply:
Good catch! Added unit tests in bd94646f66fce9b11dd3b302e7836a09b4328a42